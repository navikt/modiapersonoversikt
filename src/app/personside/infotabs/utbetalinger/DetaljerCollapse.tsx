import * as React from 'react';
import { ReactNode } from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { UnmountClosed } from 'react-collapse';
import { FlexEnd } from '../../../../components/common-styled-components';
import NavFrontendChevron from 'nav-frontend-chevron';
import { Normaltekst } from 'nav-frontend-typografi';

interface Props {
    open: boolean;
    toggle: () => void;
    children: ReactNode;
}

const Wrapper = styled<{ open: boolean }, 'div'>('div')`
  transition: .5s;
  padding: 0 ${theme.margin.px20} ${theme.margin.px20};
  ${props => props.open && theme.ekspandert};
`;

const CollapseAnimasjon = styled<{ open: boolean }, 'div'>('div')`
  transition: .5s;
  padding: 0 ${theme.margin.px20};
  ${props => props.open && 'padding: 0'};
`;

const KnappWrapper = styled<{ open: boolean }, 'button'>('button')`
  border: none;
  padding: .1rem .2rem;
  margin-top: ${props => props.open && '1rem'};
  border-radius: ${theme.borderRadius.knapp};
  cursor: pointer;
  background-color: transparent;
  display: flex;
  align-items: center;
  color: ${theme.color.lenke};
  &:focus {
    ${theme.focus}
  }
  &:hover {
    opacity: 0.8;
    text-decoration: underline;
  }
 @media print {
    display: none;
 }
`;

const Luft = styled.span`
  margin-right: .5rem;
`;

interface KnappProps {
    onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
    open: boolean;
}

function DetaljerKnapp(props: KnappProps) {
    return (
        <KnappWrapper
            onClick={props.onClick}
            aria-expanded={props.open}
            open={props.open}
        >
            <Luft>
                <Normaltekst tag="span">
                    {props.open ? 'Skjul' : 'Vis'} detaljer
                </Normaltekst>
            </Luft>
            <NavFrontendChevron type={props.open ? 'opp' : 'ned'}/>
        </KnappWrapper>
    );
}

function DetaljerCollapse(props: Props) {
    return (
        <Wrapper open={props.open}>
            <CollapseAnimasjon open={props.open}>
                <UnmountClosed isOpened={props.open}>
                    {props.children}
                </UnmountClosed>
            </CollapseAnimasjon>
            <FlexEnd>
                <DetaljerKnapp
                    onClick={props.toggle}
                    open={props.open}
                />
            </FlexEnd>
        </Wrapper>
    );
}

export default DetaljerCollapse;
