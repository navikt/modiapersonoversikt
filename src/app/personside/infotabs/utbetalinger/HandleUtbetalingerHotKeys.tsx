import * as React from 'react';
import { Utbetaling, Ytelse } from '../../../../models/utbetalinger';
import { flatMapYtelser } from './utils/utbetalingerUtils';
import { AnyAction, Dispatch } from 'redux';
import {
    setEkspanderYtelse,
    setNyYtelseIFokus
} from '../../../../redux/utbetalinger/utbetalingerStateReducer';
import { AppState } from '../../../../redux/reducers';
import { connect } from 'react-redux';
import styled from 'styled-components';

interface OwnProps {
    children: React.ReactNode;
    utbetalinger: Utbetaling[];
}

interface DispatchProps {
    settYtelseIFokus: (ytelse: Ytelse | null) => void;
    ekspanderYtelse: (ytelse: Ytelse, ekspander: boolean) => void;
}

interface StateProps {
    ytelseIFokus: Ytelse | null;
    ekspanderteYtelser: Ytelse[];
}

type Props = DispatchProps & OwnProps & StateProps;

function eventTargetIsButton(event: React.KeyboardEvent) {
    return event.target instanceof HTMLButtonElement;
}

const Wrapper = styled.div`
  &:focus {
    outline: none;
  }
`;

class HandleUtbetalingerHotKeys extends React.Component<Props> {

    constructor(props: Props) {
        super(props);
        this.handleKeyDown = this.handleKeyDown.bind(this);
    }

    handleKeyDown(event: React.KeyboardEvent) {
        switch (event.which) {
            case 13:
                this.handleEnter(event);
                break;
            case 32:
                this.handleSpace(event);
                break;
            case 38:
                this.settForrigeYtelseIFokus();
                break;
            case 40:
                this.settNesteYtelseIFokus();
                break;
            default:
        }
    }

    settNesteYtelseIFokus() {
        const ytelser: Ytelse[] = flatMapYtelser(this.props.utbetalinger);
        const currentIndex = this.props.ytelseIFokus ? ytelser.indexOf(this.props.ytelseIFokus) : -1;
        const nesteYtelse = ytelser[currentIndex + 1] || ytelser[0];
        this.props.settYtelseIFokus(nesteYtelse);
    }

    settForrigeYtelseIFokus() {
        const ytelser: Ytelse[] = flatMapYtelser(this.props.utbetalinger);
        const currentIndex = this.props.ytelseIFokus ? ytelser.indexOf(this.props.ytelseIFokus) : ytelser.length;
        const forrigeYtelse = ytelser[currentIndex - 1] || ytelser[ytelser.length - 1];
        this.props.settYtelseIFokus(forrigeYtelse);
    }

    handleEnter(event: React.KeyboardEvent) {
        if (event.repeat || eventTargetIsButton(event)) {
            return;
        }
        this.toggleEkspanderYtelseIFokus();
    }

    toggleEkspanderYtelseIFokus() {
        const ytelseIFokus = this.props.ytelseIFokus;
        if (!ytelseIFokus) {
            return;
        }
        const erEkspandert = this.props.ekspanderteYtelser.includes(ytelseIFokus);
        this.props.ekspanderYtelse(ytelseIFokus, !erEkspandert);
    }

    handleSpace(event: React.KeyboardEvent) {
        if (eventTargetIsButton(event)) {
            return;
        }
        event.preventDefault();
        this.toggleEkspanderYtelseIFokus();
    }

    render() {
        return (
            <Wrapper onKeyDown={this.handleKeyDown} tabIndex={-1}>
                {this.props.children}
            </Wrapper>
        );
    }
}

function mapDispatchToProps(dispatch: Dispatch<AnyAction>): DispatchProps {
    return {
        settYtelseIFokus: ytelse => dispatch(setNyYtelseIFokus(ytelse)),
        ekspanderYtelse: (ytelse: Ytelse, ekspander: boolean) => dispatch(setEkspanderYtelse(ytelse, ekspander))
    };
}

function mapStateToProps(state: AppState): StateProps {
    return {
        ytelseIFokus: state.utbetalinger.ytelseIFokus,
        ekspanderteYtelser: state.utbetalinger.ekspanderteYtelser
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(HandleUtbetalingerHotKeys);
