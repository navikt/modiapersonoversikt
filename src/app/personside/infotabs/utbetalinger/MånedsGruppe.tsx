import * as React from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { ArrayGroup } from '../../../../utils/groupArray';
import { Normaltekst } from 'nav-frontend-typografi';
import { Bold, Uppercase } from '../../../../components/common-styled-components';
import { getGjeldendeDatoForUtbetaling } from './utils/utbetalingerUtils';
import UtbetalingsKomponent from './utbetaling/Utbetaling';
import { Utbetaling } from '../../../../models/utbetalinger';

interface Props {
    gruppe: ArrayGroup<Utbetaling>;
}

const Wrapper = styled.li`
  > *:first-child {
    background-color: ${theme.color.kategori};
    padding: .2rem ${theme.margin.px20};
  }
  ol {
    padding: 0;
    margin: 0;
  }
  ol > *:not(:first-child) {
    border-top: ${theme.border.skille};
  }
`;

function Månedsgruppe({gruppe}: Props) {
    const utbetalingsKomponenter = gruppe.array.map(utbetaling => (
        <UtbetalingsKomponent
            key={getGjeldendeDatoForUtbetaling(utbetaling) + utbetaling.nettobeløp}
            utbetaling={utbetaling}
        />
    ));
    return (
        <Wrapper>
            <Normaltekst tag={'h3'}><Bold><Uppercase>{gruppe.category}</Uppercase></Bold></Normaltekst>
            <ol>
                {utbetalingsKomponenter}
            </ol>
        </Wrapper>
    );
}

export default Månedsgruppe;
