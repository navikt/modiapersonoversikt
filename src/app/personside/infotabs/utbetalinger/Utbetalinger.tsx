import * as React from 'react';
import { Utbetaling, UtbetalingerResponse } from '../../../../models/utbetalinger';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { Normaltekst, Undertittel } from 'nav-frontend-typografi';
import TotaltUtbetalt from './totalt utbetalt/TotaltUtbetalt';
import { ArrayGroup, groupArray, GroupedArray } from '../../../../utils/groupArray';
import {
    fjernTommeUtbetalinger,
    getTypeFromYtelse,
    månedOgÅrForUtbetaling,
    reduceUtbetlingerTilYtelser,
    utbetalingDatoComparator,
    utbetaltTilBruker
} from './utils/utbetalingerUtils';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import Månedsgruppe from './MånedsGruppe';
import HandleUtbetalingerArrowKeys from './HandleUtbetalingerHotKeys';
import AriaNotification from '../../../../components/AriaNotification';
import { UtbetalingFilterState } from '../../../../redux/utbetalinger/types';

const UtbetalingerArticle = styled.article`
    ${theme.hvittPanel};
    margin-top: ${theme.margin.layout};
    margin-bottom: ${theme.margin.layout};
    > *:first-child {
        padding: ${theme.margin.px20};
    }
`;

const UtbetalingerListe = styled.ol`
    padding: 0;
    margin: 0;
`;

const Wrapper = styled.div`
    ol {
        list-style: none;
    }
`;

const OnOneLine = styled.div`
    display: inline-flex;
    align-items: center;
    > *:last-child {
        margin-left: 1rem;
    }
`;

export function getFiltrerteUtbetalinger(utbetalinger: Utbetaling[], filter: UtbetalingFilterState) {
    return utbetalinger
        .filter(utbetaling => filtrerPaUtbetaltTilValg(utbetaling, filter))
        .map(utbetaling => filtrerBortYtelserSomIkkeErValgt(utbetaling, filter))
        .filter(fjernTommeUtbetalinger);
}

function filtrerPaUtbetaltTilValg(utbetaling: Utbetaling, filter: UtbetalingFilterState) {
    return filter.utbetaltTil.includes(utbetaling.erUtbetaltTilPerson ? utbetaltTilBruker : utbetaling.utbetaltTil);
}

function filtrerBortYtelserSomIkkeErValgt(utbetaling: Utbetaling, filter: UtbetalingFilterState): Utbetaling {
    const ytelser = reduceUtbetlingerTilYtelser([utbetaling]).filter(ytelse =>
        filter.ytelser.includes(getTypeFromYtelse(ytelse))
    );
    return {
        ...utbetaling,
        ytelser: ytelser
    };
}

interface UtbetalingerProps {
    utbetalingerData: UtbetalingerResponse;
    filter: UtbetalingFilterState;
}

function Utbetalinger({ filter, ...props }: UtbetalingerProps) {
    const filtrerteUtbetalinger = getFiltrerteUtbetalinger(props.utbetalingerData.utbetalinger, filter);
    if (filtrerteUtbetalinger.length === 0) {
        return (
            <div role="alert">
                <AlertStripeInfo>
                    Det finnes ingen utbetalinger for valgte kombinasjon av periode og filtrering.
                </AlertStripeInfo>
            </div>
        );
    }

    const utbetalingerGruppert: GroupedArray<Utbetaling> = groupArray(
        filtrerteUtbetalinger.sort(utbetalingDatoComparator),
        månedOgÅrForUtbetaling
    );

    const månedsGrupper = utbetalingerGruppert.map((gruppe: ArrayGroup<Utbetaling>) => (
        <Månedsgruppe key={gruppe.category} gruppe={gruppe} {...props} />
    ));

    return (
        <Wrapper>
            <AriaNotification
                beskjed={`Det finnes ${filtrerteUtbetalinger.length} utbetalinger for valgt periode og filtrering`}
            />
            <TotaltUtbetalt utbetalinger={filtrerteUtbetalinger} periode={props.utbetalingerData.periode} />
            <HandleUtbetalingerArrowKeys utbetalinger={filtrerteUtbetalinger}>
                <UtbetalingerArticle aria-label="Utbetalinger">
                    <OnOneLine>
                        <Undertittel>Utbetalinger</Undertittel>
                        <Normaltekst>({filtrerteUtbetalinger.length} utbetalinger)</Normaltekst>
                    </OnOneLine>
                    <UtbetalingerListe aria-label="Måneder">{månedsGrupper}</UtbetalingerListe>
                </UtbetalingerArticle>
            </HandleUtbetalingerArrowKeys>
        </Wrapper>
    );
}

export default Utbetalinger;
