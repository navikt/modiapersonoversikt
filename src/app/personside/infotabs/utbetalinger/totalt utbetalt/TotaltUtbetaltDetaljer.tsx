import * as React from 'react';
import { Fragment } from 'react';
import styled from 'styled-components';
import { TotaltUtbetaltProps } from './TotaltUtbetalt';
import { Normaltekst } from 'nav-frontend-typografi';
import { Utbetaling, Ytelse, Ytelseskomponent } from '../../../../../models/utbetalinger';
import { groupArray } from '../../../../../utils/groupArray';
import theme from '../../../../../styles/personOversiktTheme';
import {
    filtrerBortUtbetalingerSomIkkeErUtbetalt,
    formaterNOK,
    getBruttoSumYtelser,
    getNettoSumYtelser,
    getPeriodeFromYtelser,
    getTrekkSumYtelser,
    getTypeFromYtelse,
    reduceUtbetlingerTilYtelser
} from '../utils/utbetalingerUtils';
import { formaterDato } from '../../../../../utils/dateUtils';
import ErrorBoundary from '../../../../../components/ErrorBoundary';
import * as moment from 'moment';
import { sorterAlfabetisk } from '../../../../../utils/string-utils';
import DetaljerCollapse from '../../../../../components/DetaljerCollapse';

interface OwnProps {
    visDetaljer: boolean;
    toggleVisDetaljer: () => void;
}

type Props = TotaltUtbetaltProps & OwnProps;

const DetaljerStyle = styled.aside`
  h2 {
    margin: 0;
    padding: ${theme.margin.px20} 0;
  }
  th:not(:first-child) {
    font-weight: normal;
  }
  th {
    text-transform: uppercase;
  }
  tr {
    display: flex;
    flex-flow: row wrap;
    position: relative; /* IE11 */
    > *:first-child {
      flex-grow: 1;
      display: block; /* IE11 */
    }
    .sumBrutto, .sumNetto, .sumTrekk {
      position: absolute; /* IE11 */
    }
    .sumBrutto {
      right: 12rem; /* IE11 */
    }
    .sumTrekk {
      right: 6rem; /* IE11 */
    }
    .sumNetto {
      right: 0; /* IE11 */
    }
    td.sumNetto {
      font-weight: bold;
    }
    .periodeForYtelse, .ytelseDetaljer {
      flex-basis: 100%;
    }
    .periodeForYtelse {
      display: block; /* IE11 */
      text-align: left;
    }
    .ytelseDetaljer {
      display: block; /* IE11 */
      flex-grow: 1;
      text-align: right;
      margin-right: 12rem;
      dl {
        display: flex;
        flex-flow: row wrap;
        dt {
          flex-basis: 50%;
          flex-grow: 1;
        }
        dd {
          padding-left: 1rem;
          flex-basis: 7rem;
        }
      }
    }
  }
  tbody tr {
    margin-top: .5rem;
    > *:nth-last-child(n+3) {
        padding-top: .5rem;
        border-top: ${theme.border.skilleSvak};
    }
  }
  @media print {
    table { page-break-inside: auto; }
    tbody { page-break-inside: auto }
  }
`;

function getAlleYtelsesKomponenterFraYtelser(ytelser: Ytelse[]) {
    return ytelser.reduce(
        (acc: Ytelseskomponent[], ytelse: Ytelse) => {
            if (!ytelse.ytelseskomponentListe) {
                throw new Error('Ytelse mangler ytelseskomponenter');
            }
            return [...acc, ...ytelse.ytelseskomponentListe];
        },
        []
    );
}

function getYtelsesKomponentSammendragListe(ytelser: Ytelse[]) {
    const alleYtelsesKomponenter = getAlleYtelsesKomponenterFraYtelser(ytelser);
    const ytelsesKomponenterGruppertPåType = groupArray(
        alleYtelsesKomponenter,
        ytelseskomponent => ytelseskomponent.ytelseskomponenttype
    );
    const listeKomponenter = ytelsesKomponenterGruppertPåType.map(gruppe => {
        const sum = gruppe.array.reduce((acc, ytelsesKomponent) => acc + ytelsesKomponent.ytelseskomponentbeløp, 0);
        return (
            <Fragment key={gruppe.category}>
                <dt>{gruppe.category}</dt>
                <dd>{formaterNOK(sum)}</dd>
            </Fragment>
        );
    });
    return <dl>{listeKomponenter}</dl>;
}

function getAlleUtbetalteYtelserFraUtbetalinger(utbetalinger: Utbetaling[]) {
    const utbetalteUtbetalinger = utbetalinger.filter(filtrerBortUtbetalingerSomIkkeErUtbetalt);
    return reduceUtbetlingerTilYtelser(utbetalteUtbetalinger);
}

function getTypeOgÅrFromYtelse(ytelse: Ytelse): string {
    return getTypeFromYtelse(ytelse) + ' ' + moment(ytelse.periode.slutt).year();
}

function getYtelserSammendrag(utbetalinger: Utbetaling[]) {
    const alleUtbetalteYtelser: Ytelse[] = getAlleUtbetalteYtelserFraUtbetalinger(utbetalinger);
    const ytelserGruppertPåTema = groupArray(
        alleUtbetalteYtelser,
        getTypeOgÅrFromYtelse
    );
    const ytelsesSammendrag = ytelserGruppertPåTema
        .sort((a, b) => sorterAlfabetisk(a.category, b.category))
        .map(gruppe => {
            const ytelser = gruppe.array;
            const ytelsesType = gruppe.category;
            const periode = getPeriodeFromYtelser(ytelser);
            const ytelsesKomponentSammendragListe = getYtelsesKomponentSammendragListe(ytelser);
            return (
                <tbody role="rowgroup" key={ytelsesType}>
                <tr role="row">
                    <th role="rowheader" scope="row">{ytelsesType}</th>
                    <td role="cell" className="sumBrutto">{formaterNOK(getBruttoSumYtelser(ytelser))}</td>
                    <td role="cell" className="sumTrekk">{formaterNOK(getTrekkSumYtelser(ytelser))}</td>
                    <td role="cell" className="sumNetto">{formaterNOK(getNettoSumYtelser(ytelser))}</td>
                    <td role="cell" className="periodeForYtelse">
                        {formaterDato(periode.fra)} - {formaterDato(periode.til)}
                    </td>
                    <td role="cell" className="ytelseDetaljer">{ytelsesKomponentSammendragListe}</td>
                </tr>
                </tbody>
            );
        });
    return ytelsesSammendrag;
}

function TotaltUtbetaltDetaljer(props: Props) {
    const ytelserSammendrag = getYtelserSammendrag(props.utbetalinger);
    return (
        <DetaljerCollapse
            open={props.visDetaljer}
            toggle={props.toggleVisDetaljer}
            tittel="sammendrag"
        >
            <ErrorBoundary>
                <DetaljerStyle aria-label="Sammendrag utbetalinger">
                    <Normaltekst tag="div">
                        <table role="table">
                            <thead role="rowgroup">
                            <tr role="row">
                                <th role="columnheader" scope="col">Ytelse</th>
                                <th role="columnheader" scope="col" className="sumBrutto">Brutto</th>
                                <th role="columnheader" scope="col" className="sumTrekk">Trekk</th>
                                <th role="columnheader" scope="col" className="sumNetto">Utbetalt</th>
                                <th role="columnheader" scope="col" className="visually-hidden">Periode</th>
                                <th role="columnheader" scope="col" className="visually-hidden">Detaljer</th>
                            </tr>
                            </thead>
                            {ytelserSammendrag}
                        </table>
                    </Normaltekst>
                </DetaljerStyle>
            </ErrorBoundary>
        </DetaljerCollapse>

    );
}

export default TotaltUtbetaltDetaljer;
