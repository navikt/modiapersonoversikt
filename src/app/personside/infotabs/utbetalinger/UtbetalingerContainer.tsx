import * as React from 'react';
import Utbetalinger from './Utbetalinger';
import { UtbetalingerResponse } from '../../../../models/utbetalinger';
import Filter from './filter/Filter';
import theme from '../../../../styles/personOversiktTheme';
import styled, { css } from 'styled-components/macro';
import ErrorBoundary from '../../../../components/ErrorBoundary';
import Arenalenke from './Arenalenke/Arenalenke';
import { BigCenteredLazySpinner } from '../../../../components/BigCenteredLazySpinner';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import { erIE11 } from '../../../../utils/erNyPersonoversikt';
import { ScrollBar, scrollBarContainerStyle } from '../utils/InfoTabsScrollBar';

const UtbetalingerStyle = styled.div`
    ${scrollBarContainerStyle(theme.media.utbetalinger.minWidth)};
    @media (min-width: ${theme.media.utbetalinger.minWidth}) {
        ${erIE11() &&
            css`
                height: 0; /* IE11 */
                flex-grow: 1; /* IE11 */
            `};
        display: flex;
        align-items: flex-start;
        > *:last-child {
            flex-grow: 1;
        }
    }
    position: relative;
`;

const FiltreringSection = styled.section`
    min-width: 19.5rem;
    flex-basis: 19.5rem;
`;

const UtbetalingerSection = styled.section`
    position: relative;
    min-width: 35rem; // Tabellene begynner Ã¥ wrappe ved bredder mindre enn dette
`;

function UtbetalingerContainer() {
    return (
        <ErrorBoundary boundaryName={'UtbetalingerContainer'}>
            <UtbetalingerStyle>
                <ScrollBar keepScrollId="utbetalinger-filter">
                    <Arenalenke />
                    <FiltreringSection>
                        <Filter />
                    </FiltreringSection>
                </ScrollBar>
                <ScrollBar keepScrollId="utbetalinger-liste">
                    <UtbetalingerSection aria-label="Filtrerte utbetalinger">
                        <RestResourceConsumer<UtbetalingerResponse>
                            getResource={restResources => restResources.utbetalinger}
                            returnOnPending={BigCenteredLazySpinner}
                        >
                            {utbetalinger => <Utbetalinger utbetalingerData={utbetalinger} />}
                        </RestResourceConsumer>
                    </UtbetalingerSection>
                </ScrollBar>
            </UtbetalingerStyle>
        </ErrorBoundary>
    );
}

export default UtbetalingerContainer;
