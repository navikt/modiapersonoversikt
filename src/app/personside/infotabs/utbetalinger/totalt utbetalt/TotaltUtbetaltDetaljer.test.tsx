import { render } from '@testing-library/react';
import { statiskMockUtbetaling } from '../../../../../mock/utbetalinger/statiskMockUtbetaling';
import type { Utbetaling, UtbetalingerPeriode } from '../../../../../models/utbetalinger';
import TotaltUtbetaltDetaljer from './TotaltUtbetaltDetaljer';

Date.now = () => new Date().getTime(); // for Ã¥ motvirke Date.now() mock i setupTests.ts

const periode: UtbetalingerPeriode = {
    startDato: '17-05-1905',
    sluttDato: '17-05-1905'
};

const mockUtbetalingUtenUtbetaltDato: Utbetaling = {
    ...statiskMockUtbetaling,
    utbetalingsdato: null
};

const mockUtbetalingReturnertForSaksbehandling: Utbetaling = {
    ...statiskMockUtbetaling,
    status: 'Returnert til NAV for saksbehandling'
};

test('Filtrerer bort utbetalinger som ikke skal medregnes og viser totalt-utbetalt-detaljer riktig', () => {
    vi.useRealTimers();
    const visittkortheader = render(
        <TotaltUtbetaltDetaljer
            visDetaljer={true}
            toggleVisDetaljer={() => null}
            periode={periode}
            utbetalinger={[
                statiskMockUtbetaling,
                statiskMockUtbetaling,
                mockUtbetalingUtenUtbetaltDato,
                mockUtbetalingReturnertForSaksbehandling
            ]}
        />
    );

    const json = visittkortheader.asFragment();
    expect(json).toMatchSnapshot();
});
