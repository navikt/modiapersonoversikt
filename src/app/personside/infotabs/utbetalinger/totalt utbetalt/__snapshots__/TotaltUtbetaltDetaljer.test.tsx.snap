// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filtrerer bort utbetalinger som ikke skal medregnes og viser totalt-utbetalt-detaljer riktig 1`] = `
Array [
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 {
  position: relative;
  border: none;
  padding: 0;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c3:after {
  border-bottom: 1px #b7b1a9 solid;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
}

.c3:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c3:hover {
  opacity: 0.8;
}

.c3:hover:after {
  content: '';
  border-color: #0067c5;
}

.c4 {
  margin-right: 0.5rem;
}

.c0 {
  margin-right: 1.25rem;
}

@media print {
  .c1 {
    display: none;
  }
}

@media print {

}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <button
          aria-expanded={true}
          className="c3"
          onClick={[Function]}
        >
          <span
            className="c4"
          >
            <span
              className="typo-normal"
            >
              Skjul sammendrag
            </span>
          </span>
          <i
            className="nav-frontend-chevron chevronboks chevron--opp"
          />
        </button>
      </div>
    </div>
  </div>,
  .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  position: relative;
  border: none;
  padding: 0;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c5:after {
  border-bottom: 1px #b7b1a9 solid;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
}

.c5:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c5:hover {
  opacity: 0.8;
}

.c5:hover:after {
  content: '';
  border-color: #0067c5;
}

.c6 {
  margin-right: 0.5rem;
}

.c0 {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  padding: 0 1.25rem 1.25rem;
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
  margin-top: 1.25rem;
}

.c1 {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  padding: 0 1.25rem;
  padding: 0;
}

.c2 h2 {
  margin: 0;
  padding: 1.25rem 0;
}

.c2 th:not(:first-child) {
  font-weight: normal;
}

.c2 th {
  text-transform: uppercase;
  font-weight: 600;
}

.c2 tr {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  position: relative;
}

.c2 tr > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: block;
}

.c2 tr .sumBrutto,
.c2 tr .sumNetto,
.c2 tr .sumTrekk {
  position: absolute;
}

.c2 tr .sumBrutto {
  right: 12rem;
}

.c2 tr .sumTrekk {
  right: 6rem;
}

.c2 tr .sumNetto {
  right: 0;
}

.c2 tr td.sumNetto {
  font-weight: 600;
}

.c2 tr .periodeForYtelse,
.c2 tr .ytelseDetaljer {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c2 tr .periodeForYtelse {
  display: block;
  text-align: left;
}

.c2 tr .ytelseDetaljer {
  display: block;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
  margin-right: 12rem;
}

.c2 tr .ytelseDetaljer dl {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c2 tr .ytelseDetaljer dl dt {
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 tr .ytelseDetaljer dl dd {
  padding-left: 1rem;
  -webkit-flex-basis: 7rem;
  -ms-flex-preferred-size: 7rem;
  flex-basis: 7rem;
}

.c2 tbody tr {
  margin-top: 0.5rem;
}

.c2 tbody tr > *:nth-last-child(n + 3) {
  padding-top: 0.5rem;
  border-top: solid 0.0625rem #b7b1a9;
}

@media print {
  .c3 {
    display: none;
  }
}

@media print {
  .c2 table {
    page-break-inside: auto;
  }

  .c2 tbody {
    page-break-inside: auto;
  }
}

<div
    className="c0"
    open={true}
  >
    <div
      className="c1"
      open={true}
    >
      <aside
        aria-label="Sammendrag utbetalinger"
        className="c2"
      >
        <div
          className="typo-normal"
        >
          <table
            role="table"
          >
            <thead
              role="rowgroup"
            >
              <tr
                role="row"
              >
                <th
                  role="columnheader"
                  scope="col"
                >
                  Ytelse
                </th>
                <th
                  className="sumBrutto"
                  role="columnheader"
                  scope="col"
                >
                  Brutto
                </th>
                <th
                  className="sumTrekk"
                  role="columnheader"
                  scope="col"
                >
                  Trekk
                </th>
                <th
                  className="sumNetto"
                  role="columnheader"
                  scope="col"
                >
                  Utbetalt
                </th>
                <th
                  className="visually-hidden"
                  role="columnheader"
                  scope="col"
                >
                  Periode
                </th>
                <th
                  className="visually-hidden"
                  role="columnheader"
                  scope="col"
                >
                  Detaljer
                </th>
              </tr>
            </thead>
            <tbody
              role="rowgroup"
            >
              <tr
                role="row"
              >
                <th
                  role="rowheader"
                  scope="row"
                >
                  Sykepenger 2017
                </th>
                <td
                  className="sumBrutto"
                  role="cell"
                >
                  40,110.00
                </td>
                <td
                  className="sumTrekk"
                  role="cell"
                >
                  -5,952.00
                </td>
                <td
                  className="sumNetto"
                  role="cell"
                >
                  34,158.00
                </td>
                <td
                  className="periodeForYtelse"
                  role="cell"
                >
                  09.11.2016
                   - 
                  31.10.2017
                </td>
                <td
                  className="ytelseDetaljer"
                  role="cell"
                >
                  <dl>
                    <dt>
                      Mouse
                    </dt>
                    <dd>
                      11,580.00
                    </dd>
                    <dt>
                      Computer
                    </dt>
                    <dd>
                      11,790.00
                    </dd>
                    <dt>
                      Bacon
                    </dt>
                    <dd>
                      16,740.00
                    </dd>
                  </dl>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </aside>
    </div>
    <div
      aria-hidden={true}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <button
            aria-expanded={true}
            className="c5"
            onClick={[Function]}
          >
            <span
              className="c6"
            >
              <span
                className="typo-normal"
              >
                Skjul sammendrag
              </span>
            </span>
            <i
              className="nav-frontend-chevron chevronboks chevron--opp"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
